governance/constitution/actions/default.js

```diff
@@ -0,0 +1,1481 @@
+  }
+}
+
+function checkNone(args) {
+  if (args !== null && args !== undefined) {
+    throw new Error(`Proposal does not accept any argument, found "${args}"`);
+  }
+}
+
+function checkEntityId(value, field) {
+  checkType(value, "string", field);
+  // This should be the hex-encoding of a SHA256 digest. This is 32 bytes long, so
+  // produces 64 hex characters.
+  const digestLength = 64;
+  if (value.length !== digestLength) {
+    throw new Error(`${field} must contain exactly ${digestLength} characters`);
+  }
+  const re = new RegExp("^[a-fA-F0-9]*$");
+  if (!re.test(value)) {
+    throw new Error(`${field} contains non-hexadecimal character`);
+  }
+}
+
+function getSingletonKvKey() {
+  // When a KV map only contains one value, this is the key at which
+  // the value is recorded
+  return new ArrayBuffer(8);
+}
+
+function getActiveRecoveryMembersCount() {
+  let activeRecoveryMembersCount = 0;
+  ccf.kv["public:ccf.gov.members.encryption_public_keys"].forEach((_, k) => {
+    let rawMemberInfo = ccf.kv["public:ccf.gov.members.info"].get(k);
+    if (rawMemberInfo === undefined) {
+      throw new Error(`Recovery member ${ccf.bufToStr(k)} has no information`);
+    }
+
+    const memberInfo = ccf.bufToJsonCompatible(rawMemberInfo);
+    if (memberInfo.status === "Active") {
+      activeRecoveryMembersCount++;
+    }
+  });
+  return activeRecoveryMembersCount;
+}
+
+function checkJwks(value, field) {
+  checkType(value, "object", field);
+  checkType(value.keys, "array", `${field}.keys`);
+  for (const [i, jwk] of value.keys.entries()) {
+    checkType(jwk.kid, "string", `${field}.keys[${i}].kid`);
+    checkType(jwk.kty, "string", `${field}.keys[${i}].kty`);
+    checkType(jwk.x5c, "array", `${field}.keys[${i}].x5c`);
+    checkLength(jwk.x5c, 1, null, `${field}.keys[${i}].x5c`);
+    for (const [j, b64der] of jwk.x5c.entries()) {
+      checkType(b64der, "string", `${field}.keys[${i}].x5c[${j}]`);
+      const pem =
+        "-----BEGIN CERTIFICATE-----\n" +
+        b64der +
+        "\n-----END CERTIFICATE-----";
+      checkX509CertBundle(pem, `${field}.keys[${i}].x5c[${j}]`);
+    }
+  }
+}
+
+function checkX509CertBundle(value, field) {
+  if (!ccf.crypto.isValidX509CertBundle(value)) {
+    throw new Error(
+      `${field} must be a valid X509 certificate (bundle) in PEM format`,
+    );
+  }
+}
+
+function invalidateOtherOpenProposals(proposalIdToRetain) {
+  const proposalsMap = ccf.kv["public:ccf.gov.proposals_info"];
+  proposalsMap.forEach((v, k) => {
+    let proposalId = ccf.bufToStr(k);
+    if (proposalId !== proposalIdToRetain) {
+      let info = ccf.bufToJsonCompatible(v);
+      if (info.state === "Open") {
+        info.state = "Dropped";
+        proposalsMap.set(k, ccf.jsonCompatibleToBuf(info));
+      }
+    }
+  });
+}
+
+function setServiceCertificateValidityPeriod(validFrom, validityPeriodDays) {
+  const rawConfig =
+    ccf.kv["public:ccf.gov.service.config"].get(getSingletonKvKey());
```

Comment:
is the key going to be all zero?

Reply:
See https://github.com/microsoft/azure-privacy-sandbox-kms/pull/242#issuecomment-2532277071